dkg/dkg.go

```diff
@@ -308,12 +308,16 @@ func startSyncProtocol(ctx context.Context, tcpNode host.Host, key *ecdsa.Privat
 	}
 
 	for _, client := range clients {
-		err := client.AwaitConnected(ctx)
+		err := client.IsConnected(ctx)
 		if err != nil {
 			return nil, err
 		}
 	}
```

Comment:
this isn't "once all peers are connected", rather this is "once all peers were connected once". 

Suggest:
 - `IsConnected` not blocking, but just return the current  status
 - refactor this to
   - check if all clients connected
   - if not, sleep for 100ms